src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -24,6 +24,12 @@ public ConfigurationRoot(IList<IConfigurationProvider> providers)
             _providers = providers;
         }
 
+        public string Key => string.Empty;
+
+        public string Path => string.Empty;
+
+        public string Value { get; set; }
```

Comment:
The main motivation of this change is mostly about having a consistent interface for traversing a configuration tree, I'm not really looking to change any behaviors (i.e. actually enabling root.Value scenarios)  so root.Value always returning null and throwing InvalidOperation on set seems reasonable